### PR TITLE
Updated ARC optimized convolution kernel to use the new TfLiteEvalTensor API.

### LINKED ISSUE #43910: Optimized arc kernels need to use the new TfLiteEvalTensor API
The TFLM team recently ported all kernels except a handful of externally maintained optimized kernels. The new API enables very low memory overhead for TFLM. The primary change is the `TfLiteTensor` C struct is only available during `TfLiteRegistration::Prepare` for a kernel. Those structs are served under temporary memory and all data is only available during the lifetime of that method. All TFLM kernels should request the `TfLiteEvalTensor` C struct during `TfLiteRegistration::Eval` calls. 

A sample change looks like this:
https://github.com/tensorflow/tensorflow/commit/e392050297b62772fb9e6aaf10cf1214cb5261e7

This issue tracks updating the Arc kernels to this new API.

@dzakhar 
@JaccovG

PTAL - happy to review any PRs as they come in.

### PR DESCRIPTION
**Pull request by gerbauz**
This pull request is part of porting ARC optimized kernels to the new TfLiteEvalTensor API.
This one is related to convolution kernel.

PR number: [1/5]

Other related PRs:
1. https://github.com/tensorflow/tensorflow/pull/44617
2. https://github.com/tensorflow/tensorflow/pull/44618
3. https://github.com/tensorflow/tensorflow/pull/44619
4. https://github.com/tensorflow/tensorflow/pull/44620
5. https://github.com/tensorflow/tensorflow/pull/44621

Fixes #43910

### CODE REVIEW

**Review by advaitjain — APPROVED** [review]

File: tensorflow/lite/micro/kernels/arc_mli/conv.cc [hunk 1]
```diff
@@ -195,14 +208,68 @@ TfLiteStatus Prepare(TfLiteContext* context, TfLiteNode* node) {
...
+    // reuse space allocated for OpData parameters
+    data->mli_weights->el_params.asym.scale.pi32 =
+        (int32_t*)data->per_channel_output_multiplier;
+    data->mli_bias->el_params.asym.scale.pi32 =
+        (int32_t*)data->per_channel_output_shift;
+
+    data->mli_weights->el_params.asym.zero_point.pi16 =
+        (int16_t*)&data->filter_zero_point;
+    data->mli_bias->el_params.asym.zero_point.pi16 =
+        (int16_t*)&data->filter_zero_point + sizeof(int16_t);
```
Comment (advaitjain) [inline tensorflow/lite/micro/kernels/arc_mli/conv.cc:232]: nit: prefer c++-style casts.

https://google.github.io/styleguide/cppguide.html#Casting

I think static_cast should do the trick here.

File: tensorflow/lite/micro/kernels/arc_mli/conv.cc [hunk 2]
```diff
@@ -195,14 +208,68 @@ TfLiteStatus Prepare(TfLiteContext* context, TfLiteNode* node) {
...
+    data->mli_in = reinterpret_cast<mli_tensor*>(
+        context->AllocatePersistentBuffer(context, sizeof(mli_tensor)));
+    data->mli_weights = reinterpret_cast<mli_tensor*>(
+        context->AllocatePersistentBuffer(context, sizeof(mli_tensor)));
+    data->mli_bias = reinterpret_cast<mli_tensor*>(
+        context->AllocatePersistentBuffer(context, sizeof(mli_tensor)));
+    data->mli_out = reinterpret_cast<mli_tensor*>(
+        context->AllocatePersistentBuffer(context, sizeof(mli_tensor)));
+    data->cfg = reinterpret_cast<mli_conv2d_cfg*>(
+        context->AllocatePersistentBuffer(context, sizeof(mli_conv2d_cfg)));
```
Comment (advaitjain) [inline tensorflow/lite/micro/kernels/arc_mli/conv.cc:221]: prefer static_cast over reinterpret_cast here

https://google.github.io/styleguide/cppguide.html#Casting

**Review by advaitjain — CHANGES_REQUESTED** [review]

File: tensorflow/lite/micro/kernels/arc_mli/conv.cc (hunk 1 above)
Comment (gerbauz) [reply tensorflow/lite/micro/kernels/arc_mli/conv.cc]: Fixed and commited.

File: tensorflow/lite/micro/kernels/arc_mli/conv.cc (hunk 2 above)
Comment (gerbauz) [reply tensorflow/lite/micro/kernels/arc_mli/conv.cc]: Fixed and committed.

**Comment by gbaned** [comment]
@gerbauz  Can you please resolve conflicts? Thanks!

**Review by advaitjain — APPROVED** [review]

**Comment by gerbauz** [comment]
> @gerbauz Can you please resolve conflicts? Thanks!
Done

**Review by advaitjain — APPROVED** [review]

**Pull request merged**